libvast/include/vast/detail/settings.hpp

```diff
@@ -27,4 +32,58 @@ void merge_settings(const caf::settings& src, caf::settings& dst,
 caf::expected<uint64_t>
 get_bytesize(caf::settings opts, std::string_view key, uint64_t defval);
 
+/// @brief Tries to extract a list value from config_value and convert it into
+/// std::vector<T>.
+/// @tparam T expected type of the list values.
+/// @param cfg_value value which should be a list type of Ts.
+/// @param error_message_on_list_extraction_failure error context of the error
+/// in case the cfg_value is not a list type.
+/// @return all list values extracted as Ts or error in 2 cases: the cfg_value
+/// is not of a list type, any value in the list is not of type T.
+template <class T>
+caf::expected<std::vector<T>>
+unpack_config_list_to_vector(const caf::config_value& cfg_value) {
+  const auto* list = caf::get_if<caf::config_value::list>(&cfg_value);
+  if (!list)
+    return caf::make_error(ec::invalid_configuration, "Failed to extract "
+                                                      "config value as list");
+
+  std::vector<T> ret;
+  ret.reserve(list->size());
+  for (const auto& e : *list) {
+    const auto* val = caf::get_if<T>(&e);
+    if (!val)
+      return caf::make_error(
+        ec::invalid_configuration,
+        fmt::format("Type mismatch while unpacking config list: expected {}, "
+                    "got {}",
+                    caf::detail::pretty_type_name(typeid(T)),
+                    caf::detail::pretty_type_name(typeid(e))));
+    ret.push_back(*val);
+  }
+
+  return ret;
+}
+
+/// @brief Tries to extract a list value from actor_system_config and convert it
+/// into std::vector<T>.
+/// @tparam T expected type of the list values.
+/// @param cfg config which contains the list value.
+/// @param cfg_list_key key holding the list value in the config.
+/// @return all list values extracted as Ts or error in 3 cases: no value exists
+/// under provided key in the config, the value in the config is not of a list
+/// type, any value in the list is not of type T.
+template <class T>
+caf::expected<std::vector<T>>
+unpack_config_list_to_vector(const caf::actor_system_config& cfg,
+                             std::string_view cfg_list_key) {
+  const auto& content = caf::content(cfg);
+  const auto it = content.find(cfg_list_key);
+  if (it == content.cend())
+    return caf::make_error(
+      ec::invalid_configuration,
+      fmt::format("No key: {} found in actor system config", cfg_list_key));
```

Comment:
```suggestion
      fmt::format("failed to find key '{}' in configuration", cfg_list_key));
```

The user doesn't care about the actor system, and then there's the usual ordering.